### PR TITLE
Implement Delete for volumes

### PR DESCRIPTION
**Pull request by ekinanp**
Supports #479.

Signed-off-by: Enis Inan <enis.inan@puppet.com>

### CODE REVIEW

File: volume/core.go [hunk 1]
```diff
@@ -29,16 +31,25 @@ type Interface interface {
 	VolumeOpen(ctx context.Context, path string) (plugin.SizedReader, error)
 	// Accepts a path and streams updates to the content associated with that path.
 	VolumeStream(ctx context.Context, path string) (io.ReadCloser, error)
+	// Deletes the volume node at the specified path. Mirrors plugin.Deletable#Delete
+	VolumeDelete(ctx context.Context, path string) (bool, error)
```
Comment (ekinanp) [inline volume/core.go]: It's unlikely that VolumeDelete will return false, but the general case is easy enough that we might as well support that capability if people ever want to use it. Also maintains symmetry w/ plugin.Deletable#Delete.

File: plugin/docker/volume.go [hunk 2]
```diff
@@ -226,6 +227,58 @@ func (v *volume) VolumeStream(ctx context.Context, path string) (io.ReadCloser,
 	return plugin.CleanupReader{ReadCloser: output, Cleanup: killAndDelete}, nil
 }
 
+func (v *volume) VolumeDelete(ctx context.Context, path string) (bool, error) {
+	// Create a container that mounts a volume and deletes its file. Run rm -rf on it.
```
Comment (MikaelSmith) [inline plugin/docker/volume.go]: This is almost entirely a duplicate of VolumeList. Can we extract it into a common helper?

A bit more work would be to extract common elements of all the Volume* commands here into a helper struct that we can safely cleanup.

File: plugin/kubernetes/pvc.go [hunk 3]
```diff
@@ -265,6 +266,42 @@ func (v *pvc) VolumeStream(ctx context.Context, path string) (io.ReadCloser, err
 	return plugin.CleanupReader{ReadCloser: output, Cleanup: delete}, nil
 }
 
+func (v *pvc) VolumeDelete(ctx context.Context, path string) (bool, error) {
+	// Create a container that mounts a pvc and inspects it. Run rm -rf on it.
```
Comment (MikaelSmith) [inline plugin/kubernetes/pvc.go]: This is also almost an exact copy of VolumeList. Only difference is the command run and that VolumeList uses the output in its return value.

**Review by MikaelSmith — CHANGES_REQUESTED** [review]

**Comment by ekinanp** [comment]
One sec, I see that VolumeOpen for pods runs cat on the pod, so am going to use the helper for that.

**Review by MikaelSmith — APPROVED** [review]

**Comment by ekinanp** [comment]
Ok, updated.

**Review by MikaelSmith — APPROVED** [review]

**Pull request merged**